### PR TITLE
🧑‍💻 Update Apple Silicon hardware, add M4 series

### PR DESCRIPTION
**Pull request by starkdmi**
Fixed:
- M2 Pro ~6.8 (FP32, 19 cores)
- M3 4.1 (FP32, 10 cores)
- M3 Pro 6.39-7.4 (FP32, 18 cores)

New:
- M4 4.6 (FP32, 10 cores)
- M4 Pro 9.2 (FP32, 20 cores)
- M4 Max 18.4 (FP32, 40 cores)

### CODE REVIEW

File: packages/tasks/src/hardware.ts
Comment (pcuenca) [inline packages/tasks/src/hardware.ts:455]: Was this wrong?

**Review by pcuenca — COMMENTED** [review]
Looks good to me! Where did you get the data?

**Comment by starkdmi** [comment]
> Looks good to me! Where did you get the data?

A few benchmarks websites plus Apple references.

**Review by pcuenca — APPROVED** [review]
Ok for me to make Apple Silicon numbers consistent and use FP32 if those figures come from ~official sources. Let's wait for another review to merge :)

**Review by julien-c — APPROVED** [review]
looks awesome! Coincidentally, I just received my M4 Max MBP so this is perfect timing:)

**Comment by julien-c** [comment]
https://www.linkedin.com/posts/julienchaumond_apple-silicon-fans-we-just-added-new-m4-activity-7267159987175440384-VAqD

**Comment by julien-c** [comment]
thx a ton for the contrib!

**Pull request merged**
